src/zope/app/container/browser/contents.py

```diff
@@ -204,6 +205,11 @@ def renameObjects(self):
         ids = request.get("rename_ids")
         newids = request.get("new_value")
 
+        if isinstance(ids, six.string_types):
+            ids = [ids]
+        if isinstance(newids, six.string_types):
+            newids = [newids]
```

Comment:
This isn't actually going through `zope.testbrowser`. It went straight from `zope.app.testing` to `zope.app.wsgi.testlayer` + `webtest.TestApp`. The tests were built on a "stateless" request-to-request model, and doesn't request form pages first before posting, which seemed like a simpler fit for `TestApp`. I'm more familiar with that API, but I didn't see a way to pass in directly a dict of object parameters and have them encoded into `x-www-urlencoded` in `zope.testbrowser`.